hibernate-reactive-core/src/test/java/org/hibernate/reactive/HqlSelectNewConstructorTest.java

```diff
@@ -0,0 +1,127 @@
+/* Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright: Red Hat Inc. and Hibernate Authors
+ */
+package org.hibernate.reactive;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import io.vertx.junit5.VertxTestContext;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Id;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.OneToMany;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import static jakarta.persistence.CascadeType.PERSIST;
+import static jakarta.persistence.FetchType.LAZY;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class HqlSelectNewConstructorTest extends BaseReactiveTest {
+
+	@Override
+	protected Collection<Class<?>> annotatedEntities() {
+		return List.of( Book.class, Author.class );
+	}
+
+	@BeforeEach
+	public void prepareDb(VertxTestContext context) {
+		Author author1 = new Author("Iain M. Banks");
+		Author author2 = new Author("Neal Stephenson");
+		Book book1 = new Book("1-85723-235-6", "Feersum Endjinn", author1);
+		Book book2 = new Book("0-380-97346-4", "Cryptonomicon", author2);
+		Book book3 = new Book("0-553-08853-X", "Snow Crash", author2);
+		test( context, getMutinySessionFactory().withTransaction( session -> session.persistAll(
+				author1, author2, book1, book2, book3
+		) ) );
+	}
+
+	@Test
+	public void testSimpleUpdateSameTransaction(VertxTestContext context) {
+		test( context, getMutinySessionFactory()
+				.withTransaction( session ->
+					session.createQuery(
+									"SELECT NEW Book(book.author) from Book book order by title desc",
+									Book.class
+							)
+							.getResultList()
+				 )
+				.invoke( books -> assertThat( books ).hasSize( 3 ) )
```

Comment:
Could you also check that the books actually contain the correct author, please?